### PR TITLE
[SYCL][NFC] Get started guide clean-up

### PR DESCRIPTION
**Pull request by bader**
- "Table of contents" is added
- A few unnecessary sections are removed
- Information for testing Intel SYCL implementation with Khronos
  SYCL conformance test suite is added
- Minor style fixes (80-char lines, markdown style, etc.)
- Link to CPU runtime is updated
- Link to Intel FPGA selector is added
- NEOGPUDeviceSelector example is fixed
- Compiler invocation example is updated - no need to link OpenCL
  library
- Visual Studio version requirement is updated

### CODE REVIEW

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 1]
```diff
@@ -1,111 +1,62 @@
...
+* [Find More](#find-more)
 
-   b. The latest/experimental Intel&reg; CPU Runtime for OpenCL&trade; Applications with SYCL support
-   on Windows `will soon be published` together with Linux runtime at [SYCL* Compiler and Runtimes](https://github.com/intel/llvm/releases)
+# Prerequisites
 
-### Get the required tools:
+* `git` - https://git-scm.com/downloads
+* `cmake` version 3.2 or later - http://www.cmake.org/download/
+* `libstdc++` version 6.0.21 or later (GCC 5.1.0 or later) -
```
Comment (romanovvlad) [inline sycl/doc/GetStartedWithSYCLCompiler.md]: Shouldn't we specify compiler required?

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 2]
```diff
@@ -1,111 +1,62 @@
...
-```bash
-cd %SYCL_HOME%
-git clone https://github.com/KhronosGroup/OpenCL-Headers
-set OPENCL_HEADERS=%SYCL_HOME%\OpenCL-Headers
-```
+**Windows**
 
-## Get `OpenCL-ICD-Loader`
-You can also find the most recent instructions for this component at https://github.com/KhronosGroup/OpenCL-ICD-Loader
+Open a developer command prompt using one of tho methods:
```
Comment (romanovvlad) [inline sycl/doc/GetStartedWithSYCLCompiler.md]:
```suggestion
Open a developer command prompt using one of two methods:
```

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 3]
```diff
@@ -1,111 +1,62 @@
...
-cd $SYCL_HOME
-git clone https://github.com/KhronosGroup/OpenCL-ICD-Loader
-cd OpenCL-ICD-Loader
-mkdir build
-cd build
-cmake -DOPENCL_ICD_LOADER_HEADERS_DIR=$OPENCL_HEADERS ..
-make
-export ICD_LIB=$SYCL_HOME/OpenCL-ICD-Loader/build/libOpenCL.so
-```
+- Click start menu and search for the command prompt. For MSVC-2017 it is
```
Comment (romanovvlad) [inline sycl/doc/GetStartedWithSYCLCompiler.md]:
```suggestion
- Click start menu and search for the "command prompt". For MSVC-2017 it is
```

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 4]
```diff
@@ -1,111 +1,62 @@
...
+  `x64 Native Tools Command Prompt for VS 2017`
+- run 'cmd' and then `"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat" x64`
```
Comment (romanovvlad) [inline sycl/doc/GetStartedWithSYCLCompiler.md]: Not sure it's clear what "run 'cmd'" means.

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 5]
```diff
@@ -117,62 +68,113 @@ cmake -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86" \
...
-After building the SYCL compiler and runtime, choose the amount of LIT testing you need and run one of the LIT tests suites shown below:
-### Linux:
+# Use SYCL toolchain
+
+## Install low level runtime
+
+To run SYCL  applications on OpenCL devices, OpenCL implementation(s) must be
+present in the system.
+
+To execute SYCL tests on Intel hardware following runtimes should be installed:
```
Comment (romanovvlad) [inline sycl/doc/GetStartedWithSYCLCompiler.md]: Probably it's better to say Intel devices or Intel accelerators.

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 6]
```diff
@@ -234,43 +236,44 @@ int main() {
...
+
+and run following command.
+
+```bash
+clang++ -fsycl simple-sycl-app.cpp -o simple-sycl-app.exe -lOpenCL
+```
 
 This `simple-sycl-app.exe` application doesn't specify SYCL device for execution,
 so SYCL runtime will first try to execute on OpenCL GPU device first, if OpenCL
 GPU device is not found, it will try to run OpenCL CPU device; and if OpenCL
```
Comment (romanovvlad) [inline sycl/doc/GetStartedWithSYCLCompiler.md]: As far as I remember it will try accelerator if CPU device is not found.

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 1 above)
Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: Do you mean: should we require GCC version instead of libstdc++?

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 3 above)
Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: Could clarify why command prompt should be in the double quotes?

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 4 above)
Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: Do you have any suggestions for new wording?

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 1 above)
Comment (romanovvlad) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: 1. I think we should list compiler in prerequisites.
2. I believe we don't need to specify libstdc++ in addition to compiler.

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 6 above)
Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: I changed wording in this section to avoid description of SYCL implementation details.

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 4 above)
Comment (romanovvlad) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: Ctrl-R, write "cmd", click enter, then run ...
Or open regular command prompt(click start and search for "Command prompt") then run ...

**Comment by bader** [comment]
@romanovvlad, I think I applied all your comments.
Please, check.

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 7]
```diff
@@ -1,111 +1,63 @@
...
+* [Find More](#find-more)
 
-   b. The latest/experimental Intel&reg; CPU Runtime for OpenCL&trade; Applications with SYCL support
-   on Windows `will soon be published` together with Linux runtime at [SYCL* Compiler and Runtimes](https://github.com/intel/llvm/releases)
+# Prerequisites
 
-### Get the required tools:
+* `git` - https://git-scm.com/downloads
+* `cmake` version 3.2 or later - http://www.cmake.org/download/
+* `GCC` version 5.1.0 or later (including libstdc++) -
```
Comment (vladimirlaz) [inline sycl/doc/GetStartedWithSYCLCompiler.md]: GCC is needed only on Linux

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 8]
```diff
@@ -1,111 +1,63 @@
...
+  https://gcc.gnu.org/install/
+* `python` - https://www.python.org/downloads/release/python-2716/
+* `Visual Studio 2017` or later (Windows only) -
+   https://visualstudio.microsoft.com/downloads/
```
Comment (vladimirlaz) [inline sycl/doc/GetStartedWithSYCLCompiler.md]: Please specify exact version (MSVC 19.15.26730.0 and higher). We are observing build failures on earlier versions.

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 9]
```diff
@@ -117,62 +69,113 @@ cmake -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86" \
...
+present in the system.
+
+To execute SYCL tests on Intel devices following runtimes should be installed:
+
+**Linux**
+
+* Intel GPU [Intel&reg; Graphics Compute Runtime for
+   OpenCL&trade;](https://github.com/intel/compute-runtime/releases)
+* Intel CPU - [Experimental Intel&reg; CPU Runtime for OpenCL&trade; Applications
+   with SYCL support](https://github.com/intel/llvm/releases/tag/2019-08)
```
Comment (vladimirlaz) [inline sycl/doc/GetStartedWithSYCLCompiler.md]: for GPU you passed the general location for CPU RT you passed exact version.
Also different approach is used on Windows and Linux. Can we have common was of specifying RTs.

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 10]
```diff
@@ -117,62 +69,113 @@ cmake -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86" \
 -DLLVM_ENABLE_PROJECTS="clang;llvm-spirv;sycl" \
 -DLLVM_EXTERNAL_SYCL_SOURCE_DIR=$SYCL_HOME/llvm/sycl \
 -DLLVM_EXTERNAL_LLVM_SPIRV_SOURCE_DIR=$SYCL_HOME/llvm/llvm-spirv \
--DOpenCL_INCLUDE_DIR="$OPENCL_HEADERS" -DOpenCL_LIBRARY="$ICD_LIB" \
 $SYCL_HOME/llvm/llvm
 
 make -j`nproc` sycl-toolchain
```
Comment (vladimirlaz) [inline sycl/doc/GetStartedWithSYCLCompiler.md]: why we have make on Linux and ninja on linux. lets use Ninja everywhere as it expected to be more effective

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 11]
```diff
@@ -234,43 +237,43 @@ int main() {
...
+**Windows**
+```bat
+set PATH=%SYCL_HOME%\build\bin;%PATH%
+set LIB=%SYCL_HOME%\build\lib;%LIB%
+```
+
+and run following command.
+
+```bash
+clang++ -fsycl simple-sycl-app.cpp -o simple-sycl-app.exe -lOpenCL
```
Comment (vladimirlaz) [inline sycl/doc/GetStartedWithSYCLCompiler.md]: -lOpenCL is not needed in compile/link command line as soon as there are no direct OpenCL call in application.

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 7 above)
Comment (AlexeySachkov) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: We probably can rewrite this into something like:

```
* Compiler:
  * For Linux: `GCC` version 5.1.0 ...
  * For Windows: `Visual Studio 2017` ...

```

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 12]
```diff
@@ -117,62 +69,113 @@ cmake -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86" \
...
+
+**Windows**
+```bat
 cd %SYCL_HOME%
 git clone https://github.com/intel/llvm -b sycl
 mkdir %SYCL_HOME%\build
 cd %SYCL_HOME%\build
 
-cmake -G "Ninja" -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86" -DLLVM_EXTERNAL_PROJECTS="llvm-spirv;sycl" -DLLVM_ENABLE_PROJECTS="clang;llvm-spirv;sycl" -DLLVM_EXTERNAL_SYCL_SOURCE_DIR="%SYCL_HOME%\llvm\sycl" -DLLVM_EXTERNAL_LLVM_SPIRV_SOURCE_DIR="%SYCL_HOME%\llvm\llvm-spirv" -DCMAKE_C_COMPILER=cl -DCMAKE_CXX_COMPILER=cl -DCMAKE_C_FLAGS="/GS" -DCMAKE_CXX_FLAGS="/GS" -DCMAKE_EXE_LINKER_FLAGS="/NXCompat /DynamicBase" -DCMAKE_SHARED_LINKER_FLAGS="/NXCompat /DynamicBase" -DOpenCL_INCLUDE_DIR="%OPENCL_HEADERS%" -DOpenCL_LIBRARY="%ICD_LIB%" "%SYCL_HOME%\llvm\llvm"
+cmake -G "Ninja" -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86"^
```
Comment (AlexeySachkov) [inline sycl/doc/GetStartedWithSYCLCompiler.md]: Why did you add `^` symbol at the and of each line here?

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 9 above)
Comment (AlexeySachkov) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: I think that is okay to put link to a general location here. Certain recommended version is changed from time to time and mentioned in release notes. So, probably it worth say that exact recommended version can be found in release notes

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 13]
```diff
@@ -234,43 +237,43 @@ int main() {
...
+export LD_LIBRARY_PATH=$SYCL_HOME/build/lib:$LD_LIBRARY_PATH
+```
+
+**Windows**
+```bat
+set PATH=%SYCL_HOME%\build\bin;%PATH%
+set LIB=%SYCL_HOME%\build\lib;%LIB%
+```
+
+and run following command.
```
Comment (AlexeySachkov) [inline sycl/doc/GetStartedWithSYCLCompiler.md]:
```suggestion
and run following command:
```

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 12 above)
Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: Do you know better ways to write multi-line commands in command prompt?

Comment (AlexeySachkov) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: Oh, I didn't realize the fact that original command was written in just a one line. No, I cannot suggest anything better

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 8 above)
Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: Could you provide Visual Studio version, please?
https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 9 above)
Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: > Also different approach is used on Windows and Linux. Can we have common was of specifying RTs.

What do you mean by "different approach"? Could you clarify, please?

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 10 above)
Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: I think @v-klochkov answered this question already in https://github.com/intel/llvm/pull/494. IIRC, `ninja` is installed with Visual Studio.
`make` is more ubiquitous on Linux, but `ninja` can be also used.

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 11 above)
Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: `-lOpenCL` is needed.

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 8 above)
Comment (v-klochkov) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: We had no problems with MSVC2017 before. If It does not work we could fix it.
CLANG requires MSVC2017 or newer, why we would raise the bar?

**Comment by agozillon** [comment]
Might be worth fixing the scheduler example while you're here so we can close this issue: https://github.com/intel/llvm/issues/22

**Comment by bader** [comment]
@vladimirlaz, @romanovvlad, please, review again to confirm that your concerns are resolved.

File: sycl/doc/GetStartedWithSYCLCompiler.md (hunk 11 above)
Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: It looks like @vladimirlaz is right.
SYCL headers do not use OpenCL API directly anymore, so there is no need to link with OpenCL library anymore!

Comment (v-klochkov) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: Do not remove -lOpenCL from here yet. I was going to do the same about a week ago, but found a case with direct call of OpenCL. I suggest run LIT + CTS without -lOpenCL first.

Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: As @vladimirlaz explained, we have a few test, using OpenCL API directly. We need to link OpenCL library explicitly only in this cases.
@v-klochkov, I'll update the lit tests according to updated instructions in a separate patch.

File: sycl/doc/GetStartedWithSYCLCompiler.md [hunk 14]
```diff
@@ -1,111 +1,64 @@
...
+**Windows**
 
-## Get `OpenCL-ICD-Loader`
-You can also find the most recent instructions for this component at https://github.com/KhronosGroup/OpenCL-ICD-Loader
+Open a developer command prompt using one of two methods:
 
-### Linux:
-```bash
-cd $SYCL_HOME
-git clone https://github.com/KhronosGroup/OpenCL-ICD-Loader
```
Comment (v-klochkov) [inline sycl/doc/GetStartedWithSYCLCompiler.md]: Alexey, I missed the fact that you removed the build steps for ICD-Loader.
I believe that was not good move. make and nmake generators are stable, I also had some success with ninja during the main SYCL compiler build, but when ICD-Loader and Headers are not pre-built before the main SYCL build, ninja again gives this error:

D:\iusers\vklochko\ws\gs_public\build_ninja2>ninja sycl-toolchain
ninja: error: 'lib/OpenCL.lib', needed by 'bin/sycl.dll', missing and no known rule to make it

Comment (AlexeySachkov) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: @v-klochkov, I encountered similar error (unfortunately, I've lost build logs) while using MSVC generator: we definitely have something misconfigured/missing

Comment (bader) [reply sycl/doc/GetStartedWithSYCLCompiler.md]: This looks like a bug in the cmake script, feel free to file a bug report!
